### ISSUE TITLE
Docker ps/images command hangs/fails

### ISSUE THREAD

**Issue by danmelton** [docker/for-mac#334]
### Expected behavior

```
docker ps 
```
or 
```
docker images
```

displays list of images/processes

### Actual behavior

Both hang

### Information

Diagnostic ID: 85406DBD-EDF3-4044-87E4-561186803DF5
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.10.5 (Build 14F1713)
[ERROR]  docker-cli
         docker ps failed
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

1. Computer shuts down due battery
2. Computer restarts when power is available
3. Docker is running 
4. Docker os or docker images hangs/fails from time out
5. Restart Docker through Docker for Mac restart UI
6. Docker os or docker images hangs/fails from time out
7. Reset docker to 'factory settings' through UI 
8. Docker ps works but have to rebuild or pull all docker images 
9: Make frown face :(, but docker still awesome, so change back to smiley face

**Comment by mchiang0610** [comment]
Merging with #119

**Comment by samoht** [comment]
Detected symptom of problem 'invariant-violated' in 85406DBD-EDF3-4044-87E4-561186803DF5/20160817-130444.

An invariant within the code of one of the components was not held. Docker needs to be restarted.

May be related to docker/for-mac#89

The following log matches:
```
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Notice>: Volume.stop 11eda43097cd427747b0284a6462f869f42920253af56929dac6bc60c02e2e61 (paths = [/Users/dan/projects/demeter/style/test/dummy/bin, /Users/dan/projects/demeter/style/lib, /Users/dan/projects/demeter/style/test/dummy/app, /Users/dan/projects/demeter/style/test/dummy/public, /Users/dan/projects/demeter/style/package.json, /Users/dan/projects/demeter/style/public, /Users/dan/projects/demeter/style/assets])
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Aug 11 06:49:45 Laptop Docker[com.docker.osxfs][17717] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
```

/cc @dsheets

**Comment by dsheets** [comment]
This invariant violation was a symptom of a use-after-free bug in the file system event watching code which occurred due to multiple containers with volume mounts racing to set up their mounts. It is possible that it caused your hangs or crashes and has been fixed in the recently released Beta 29.3. Please open a new issue for any further problems. Thanks!
